dask/array/core.py

```diff
@@ -610,13 +616,32 @@ def map_blocks(
             'chunk-shape': (100,),
             'dtype': dtype('float64')}}
 
-    For each argument and keyword arguments that are dask arrays (the positions
-    of which are the first index), you will receive the shape of the full
-    array, the number of chunks of the full array in each dimension, the chunk
-    location (for example the fourth chunk over in the first dimension), and
-    the array location (for example the slice corresponding to ``40:50``). The
-    same information is provided for the output, with the key ``None``, plus
-    the shape and dtype that should be returned.
+    The keys to the ``block_info`` dictionary indicate which is the input and
+    output Dask array:
+    - **Input Dask array(s):** ``block_info[0]`` refers to the first input Dask array.
+      The dictionary key is ``0`` because that is the argument index corresponding
+      to the first input Dask array.
+      In cases where multiple Dask arrays have been passed as input to the function,
+      you can access them with the number corresponding to the input argument,
+      eg: ``block_info[1]``, ``block_info[2]``, etc.
+      (Note that if you pass multiple Dask arrays as input to the function,
+      the arrays must have the same number of chunks.)
```

Comment:
@GenevieveBuckley

Here I think it is important to add the phrase "up to broadcasting rules".

Reply:
Good point. It looks like the rule is that all subsequent Dask arrays must be able to broadcast to the first Dask array. (But you can't do it the other way around, and expect the first Dask array to broadcast to match the others).

Reply:
The sentence now says:

*"(Note that if you pass multiple Dask arrays as input to the function, the arrays must have the same number of chunks, or be able to broadcast to the shape of the first Dask array.)"*

Reply:
@GenevieveBuckley 

Perhaps I misunderstand you.  But the term "broadcasting", as used here, applies to the "block-arrays", not the arrays themselves.  A "block-array" (this is not a term in widespread use: I'm only using it here for lack of a better moniker) is effectively the array obtained by considering its blocks, or chunks, as the array elements.  The shape of such a "block-array" is what is stored in the actual array's `numblocks` attribute.  Hence the broadcasting rules apply not to the `.shape`'s of the arrays but rather to their `.numblocks`.  This makes sense, since `map_blocks` is expected to map (and therefore broadcast) blocks.

So in the following, the first "block-array" broadcasts into the second and it works:

```python
arr = da.random.random((100,100), chunks=(50, 100))	# arr.numblocks = (2, 1)
arr2 = da.random.random((50, 50), chunks=(25, 25))	# arr2.numblocks = (2, 2)
```
The converse is also true:
```python
arr = da.random.random((100,100), chunks=(50, 50))	# arr.numblocks = (2, 2)
arr2 = da.random.random((50, 50), chunks=(25, 50))	# arr2.numblocks = (2, 1)
```
And so are many other cases:
```python
arr = da.random.random((100,100), chunks=(100, 50))	# arr.numblocks = (1, 2)
arr2 = da.random.random((50, 50), chunks=(25, 50))	# arr2.numblocks = (2, 1)
```

```python
arr = da.random.random((100,100), chunks=(50, 100))	# arr.numblocks = (2, 1)
arr2 = da.random.random((50, 50), chunks=(50, 25))	# arr2.numblocks = (1, 2)
```

You can further experiment to check whether what I'm saying holds.  In fact, the `block_info` parameter's `chunk_location` values are interesting to look at to reveal how each block of each array was associated with each block of the other arrays.  

As I'm writing this I'm beginning to realize that the docs (perhaps too hastily) assume that the reader has already grasped the context of what it is attempting to describe, thus unwittingly creates uncertainty of meaning in the reader's mind.